gradio/routes.py

```diff
@@ -125,17 +125,21 @@
     files("gradio").joinpath("templates", "frontend", "assets").as_posix(),  # type: ignore
 )
 VERSION = get_package_version()
-XSS_VULNERABLE_EXTENSIONS = [
-    ".html",
-    ".htm",
-    ".js",
-    ".php",
-    ".asp",
-    ".aspx",
-    ".jsp",
-    ".xml",
-    ".svg",
-]
+XSS_SAFE_MIMETYPES = {
+    "image/jpeg",
+    "image/png",
+    "image/gif",
+    "image/webp",
+    "audio/mpeg",
+    "audio/wav",
+    "audio/ogg",
+    "video/mp4",
+    "video/webm",
+    "video/ogg",
+    "text/plain",
+    "application/pdf",
```

Comment:
pdf files can run js so best to keep out of this list for extra safety.

Otherwise looks good to me!